### DISCUSSION TITLE
Return self.flattenedPages as iterator instead of list

### DISCUSSION THREAD

**Discussion by ninnzz** [py-pdf/pypdf#994] [Ideas]
I've noticed that the `self. flattenedPages` is returned as a list.
For several purposes, I think returning it as a iterator is better than just a pure list.

- More intuitive in accessing the pdf results
```python
pages = pdf.flattenedPages
next(pages)
```

This is just a suggestion. What do you think?

**Comment by MartinThoma** [comment]
@ninnzz I don't see any place where we iterate over `flattened_pages` (it was renamed). How would this improve things?
